### PR TITLE
Issue 63: StaticView.add_nearest_neighbours() should not duplicate relationships

### PR DESCRIPTION
**Pull request by yt-ms**
* [X] fix https://github.com/Midnighter/structurizr-python/issues/63, https://github.com/Midnighter/structurizr-python/issues/40
* [X] description of feature/fix
* [X] tests added/passed
* [X] add an entry to the [next release](../CHANGELOG.rst)

`add_nearest_neighbours()` was calling `_add_element()` with `add_relationships=True` which was resulting in the duplication.  As the relationships are added explicitly based on the type filter, then simply setting this to `False` solves the problem.  Also added a check to `View` to protect against other cases where someone may be unknowingly adding the same relationship twice.

Got caught out by `flake8` yet again, so added this to `make qa` and updated the examples to be compliant.

Also fixed issue 40 whilst in the area, so diagrams with blank descriptions in the Structurizr UI can now be loaded through the Python API.  Not discovered any other similar incompatibilities yet.

--------------------

THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE Apache License v.2.0. YOU MAY OBTAIN A COPY OF THE LICENSE AT https://www.apache.org/licenses/LICENSE-2.0.

THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,6 +8,7 @@
 qa:
 	isort src/structurizr/ tests/ examples/ setup.py
 	black src/structurizr/ tests/ examples/ setup.py
+	flake8 src/structurizr/ tests/ examples/ setup.py
```
Comment (Midnighter) [inline Makefile]: I will remove this line before merging. `make qa` is intended for applying style fixes and not for checking for problems. You can either use `tox -e flake8` for that or create a separate command `make qc`.

File: examples/big_bank.py [hunk 2]
```diff
@@ -152,14 +152,16 @@ def create_big_bank_workspace():
     single_page_application.tags.add(web_browser_tag)
     mobile_app = internet_banking_system.add_container(
         "Mobile App",
-        "Provides a limited subset of the Internet banking functionality to customers via their mobile device.",
+        "Provides a limited subset of the Internet banking functionality to "
+        + "customers via their mobile device.",
```
Comment (Midnighter) [inline examples/big_bank.py]: As this is within parentheses, string concatenation should happen automatically.

```suggestion
        "customers via their mobile device.",
```

File: examples/big_bank.py [hunk 3]
```diff
@@ -152,14 +152,16 @@ def create_big_bank_workspace():
...
         "Xamarin",
         id="mobileApp",
     )
     mobile_app.tags.add(mobile_app_tag)
     web_application = internet_banking_system.add_container(
         "Web Application",
-        "Delivers the static content and the Internet banking single page application.",
+        "Delivers the static content and the Internet banking single page "
+        + "application.",
```
Comment (Midnighter) [inline examples/big_bank.py]:
```suggestion
        "Delivers the static content and the Internet banking single page "
        "application.",
```

File: examples/big_bank.py [hunk 4]
```diff
@@ -171,7 +173,8 @@ def create_big_bank_workspace():
     )
     database = internet_banking_system.add_container(
         "Database",
-        "Stores user registration information, hashed authentication credentials, access logs, etc.",
+        "Stores user registration information, hashed authentication credentials, "
+        + "access logs, etc.",
```
Comment (Midnighter) [inline examples/big_bank.py]:
```suggestion
        "Stores user registration information, hashed authentication credentials, "
        "access logs, etc.",
```

File: examples/big_bank.py [hunk 5]
```diff
@@ -211,7 +215,8 @@ def create_big_bank_workspace():
     )
     security_component = api_application.add_component(
         name="Security Component",
-        description="Provides functionality related to signing in, changing passwords, etc.",
+        description="Provides functionality related to signing in, changing passwords, "
+        + "etc.",
```
Comment (Midnighter) [inline examples/big_bank.py]:
```suggestion
        description="Provides functionality related to signing in, changing passwords, "
        "etc.",
```

File: examples/big_bank.py [hunk 6]
```diff
@@ -418,20 +425,24 @@ def create_big_bank_workspace():
...
+    # dynamicView.Add(securityComponent, "select * from users where username = ?",
+    #   database)
     # dynamicView.PaperSize = PaperSize.A5_Landscape
 
     development_deployment_view = views.create_deployment_view(
         software_system=internet_banking_system,
         key="DevelopmentDeployment",
-        description="An example development deployment scenario for the Internet Banking System.",
+        description="An example development deployment scenario for the Internet "
+        + "Banking System.",
```
Comment (Midnighter) [inline examples/big_bank.py]:
```suggestion
        description="An example development deployment scenario for the Internet "
        "Banking System.",
```

File: examples/big_bank.py [hunk 7]
```diff
@@ -440,7 +451,8 @@ def create_big_bank_workspace():
     live_deployment_view = views.create_deployment_view(
         software_system=internet_banking_system,
         key="LiveDeployment",
-        description="An example live deployment scenario for the Internet Banking System.",
+        description="An example live deployment scenario for the Internet Banking "
+        + "System.",
```
Comment (Midnighter) [inline examples/big_bank.py]:
```suggestion
        description="An example live deployment scenario for the Internet Banking "
        "System.",
```

**Review by Midnighter — CHANGES_REQUESTED** [review]
Only some style changes. Otherwise this is a great addition.

File: Makefile (hunk 1 above)
Comment (yt-ms) [reply Makefile]: OK, sure.

**Review by Midnighter — APPROVED** [review]
👍🏼

**Pull request merged**